### ISSUE TITLE
EF_RISCV_FLOAT_ABI_SOFT

### ISSUE THREAD

**Issue by ebahapo** [riscv/riscv-elf-psabi-doc#143]
The constant `EF_RISCV_FLOAT_ABI_SOFT` is not defined.  It should be added to the ELF header section as part of `e_flags` to have a value of zero.

**Comment by aswaterman** [comment]
Can you prepare a PR?

**Comment by jim-wilson** [comment]
Fixed via #144.

I see that we already have EF_RISCV_FLOAT_ABI_SOFT defined in binutils and glibc header files.  So it is an oversight that it wasn't in the psabi.  Someone should probably check llvm/lld though, I don't keep a copy of it handy, and not sure if it has its own header files for this kind of stuff.

**Comment by aswaterman** [comment]
I also changed the text to make it more obvious that EF_RISCV_FLOAT_* aren't individual bitfields, but are different values that the same two-bit field might assume.

@jim-wilson do you think the EF_RISCV_FLOAT_ABI macro, used for tests like `(e_flags & EF_RISCV_FLOAT_ABI) == EF_RISCV_FLOAT_ABI_SOFT`, should be considered part of the standard as well?

**Comment by ebahapo** [comment]
> I also changed the text to make it more obvious that EF_RISCV_FLOAT_* aren't individual bitfields, but are different values that the same two-bit field might assume.

A bit unusual notation from other psABIs that I worked with, though.

**Comment by jim-wilson** [comment]
Since both binutils and glibc already have it, it makes sense to add to the psabi too.

**Comment by ebahapo** [comment]
> I see that we already have EF_RISCV_FLOAT_ABI_SOFT defined in binutils and glibc header files. So it is an oversight that it wasn't in the psabi. Someone should probably check llvm/lld though, I don't keep a copy of it handy, and not sure if it has its own header files for this kind of stuff.

It's already defined in LLVM land as well.

**Comment by aswaterman** [comment]
@ebahapo I'm actually going to revert my commit, because I now see the point was to define the value of the macro, not explain its use.

I'll add EF_RISCV_FLOAT_ABI.

**Comment by ebahapo** [comment]
> I'll add EF_RISCV_FLOAT_ABI.

In some psABIs, bit field masks get the suffix `_MASK` or some such.

**Comment by aswaterman** [comment]
Makes sense, though I think changing it would be more pain than it's worth at this point, since it's part of binutils.

**Comment by lenary** [comment]
LLVM's ELF definitions for RISC-V already define `EF_RISCV_FLOAT_ABI_SOFT` and `EF_RISCV_FLOAT_ABI` (the latter is the mask).

**Comment by asb** [comment]
I think the originally reported issue is resolved, and there's no current plan to rename the `EF_RISCV_FLOAT_ABI` macro (which is really a hint to implementers anyway), so this issue can be closed?
